### PR TITLE
Fix generated plugin accessor compilation failure

### LINKED ISSUE #17619: Kotlin precompiled script plugins fail to compile when plugins declared in inner classes are on the classpath
Within a precompiled script plugin in `buildSrc`, the code generation for plugin accessors generates an invalid documentation comment.

I've experienced this issue when working with the https://github.com/diffplug/goomph plugin suite, due to several plugin IDs referring to inner classes implementing the `Plugin` interface -- the error below provides specifics.

### Expected Behavior
The KDoc comments for plugin accessors should not cause the `buildSrc` project to fail to compile.

### Current Behavior
The `:buildSrc:compilePluginsBlocks` task fails:

<details>
<summary>Log</summary>

```
> Task :buildSrc:compilePluginsBlocks FAILED
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:167:126: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:175:132: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:183:120: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:191:118: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:199:126: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:207:120: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:231:103: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:255:112: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:279:100: Closing bracket expected
e: C:\Users\Zach\dev\gradle-script-plugin-codegen-bug\buildSrc\build\generated-sources\kotlin-dsl-external-plugin-spec-builders\kotlin\gradle\kotlin\dsl\plugins\_6146be3f585cc77316cb14c3baab0f9d\PluginSpecBuilders.kt:303:108: Closing bracket expected

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildSrc:compilePluginsBlocks'.
> Script compilation errors:

    Line 167:  * The `com.diffplug.gradle.eclipse.buildproperties` plugin implemented by [com.diffplug.gradle.eclipse.BuildPropertiesPlugin$Legacy].

                   ^ Closing bracket expected

    Line 175:  * The `com.diffplug.gradle.eclipse.excludebuildfolder` plugin implemented by [com.diffplug.gradle.eclipse.ExcludeBuildFolderPlugin$Legacy].

                         ^ Closing bracket expected

    Line 183:  * The `com.diffplug.gradle.eclipse.mavencentral` plugin implemented by [com.diffplug.gradle.eclipse.MavenCentralPlugin$Legacy].

             ^ Closing bracket expected

    Line 191:  * The `com.diffplug.gradle.eclipse.projectdeps` plugin implemented by [com.diffplug.gradle.eclipse.ProjectDepsPlugin$Legacy].

           ^ Closing bracket expected

    Line 199:  * The `com.diffplug.gradle.eclipse.resourcefilters` plugin implemented by [com.diffplug.gradle.eclipse.ResourceFiltersPlugin$Legacy].

                   ^ Closing bracket expected

    Line 207:  * The `com.diffplug.gradle.equinoxlaunch` plugin implemented by [com.diffplug.gradle.eclipserunner.EquinoxLaunchPlugin$Legacy].

             ^ Closing bracket expected

    Line 231:  * The `com.diffplug.gradle.oomph.ide` plugin implemented by [com.diffplug.gradle.oomph.OomphIdePlugin$Legacy].
                                                                                                                    ^ Closing bracket expected

    Line 255:  * The `com.diffplug.gradle.osgi.bndmanifest` plugin implemented by [com.diffplug.gradle.osgi.BndManifestPlugin$Legacy].

     ^ Closing bracket expected

    Line 279:  * The `com.diffplug.gradle.p2.asmaven` plugin implemented by [com.diffplug.gradle.p2.AsMavenPlugin$Legacy].
                                                                                                                 ^ Closing bracket expected

    Line 303:  * The `com.diffplug.gradle.swt.nativedeps` plugin implemented by [com.diffplug.gradle.swt.NativeDepsPlugin$Legacy].

 ^ Closing bracket expected

  10 errors

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 27s
```
</details>

### Context
This prevents me from referring to any plugins in the `com.diffplug:goomph` module within my project's precompiled script plugins.

### Steps to Reproduce 

See the buildSrc project from https://github.com/zml2008/gradle-script-plugin-codegen-bug

### Your Environment
Reproducible on Gradle 6.8.3 and 7.1.1

### PR DESCRIPTION
**Pull request by donat**
For internal classes the generated accessor contained a similiar comment:

    /**
     * The `my.plugin.id` plugin implemented by [my.Plugin$Impl].
     */

The dollar sign is an invalid character and should be replaced with a dot.

I've manually verified that this PR fixes #17619.

### CODE REVIEW

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test rFM

**Comment by donat** [comment]
@bot-gradle test rFM

File: subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/PluginAccessorsClassPath.kt [hunk 1]
```diff
@@ -333,7 +333,7 @@ fun BufferedWriter.appendSourceCodeForPluginAccessors(
                     format(
                         """
                         /**
-                         * The `$id` plugin implemented by [$implementationClass].
+                         * The `$id` plugin implemented by [${implementationClass.replace('$', '.')}].
```
Comment (eskatos) [inline subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/PluginAccessorsClassPath.kt:336]: :lipstick: I would move that computation outside of the string template.

`implementationClass` is a _binary name_ of a class. This is about converting from a _binary name_ to a _source name_ for that class.

We have an existing similar function at https://github.com/gradle/gradle/blob/f6eeed6aaf58643658b61593de8e4e95b17d2c59/subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/codegen/ApiTypeProvider.kt#L571

Comment (donat) [reply subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/PluginAccessorsClassPath.kt]: Fixed. PTAL.

**Comment by donat** [comment]
@bot-gradle test rFM

**Pull request merged**